### PR TITLE
perf(debounce): remove tryCatch/errorObject for custom tryCatching

### PR DESCRIPTION
**Pull request by kwonoj**
This PR applies approach to remove custom trycatcher on debounce operator.

Numbers are

```
before
debounce - immediate |  5,530 (±7.51%) |      23,628 (±3.99%) |   4.27x |   327.3%

after
debounce - immediate |  5,593 (±9.91%) |      27,302 (±2.89%) |   4.88x |   388.1%
```

not sure if this can be considered noticeable improvement.

### CODE REVIEW

**Comment by kwonoj** [comment]
I'll check this in around later today if there isn't any suggestion.

File: src/operator/debounce.ts [hunk 1]
```diff
@@ -46,21 +43,25 @@ class DebounceSubscriber<T, R> extends OuterSubscriber<T, R> {
...
-      subscription = subscribeToResult(this, duration);
-      if (!subscription.isUnsubscribed) {
-        this.add(this.durationSubscription = subscription);
-      }
+    try {
+      duration = this.durationSelector.call(this, value);
+    } catch (err) {
+      this.destination.error(err);
+      return;
+    }
```
Comment (luisgabriel) [inline src/operator/debounce.ts]: I think @blesh's suggestion on [#1266 (comment)](https://github.com/ReactiveX/RxJS/pull/1266#discussion_r51300242) also applies here. (To move the try/catch block to a separate function.)

Comment (blesh) [inline src/operator/debounce.ts]: That's correct. A function that had the bare minimum try catch implementation in it. All other logic in other functions.

Comment (kwonoj) [inline src/operator/debounce.ts]: Got it, will update PR accordingly.

**Comment by kwonoj** [comment]
Updated PR to separate trycatch as function.

**Comment by kwonoj** [comment]
Merged with 90bf3f1.

**Pull request closed without merging**